entity-service/src/main/kotlin/com/egm/stellio/entity/web/EntityHandler.kt

```diff
@@ -113,7 +139,13 @@ class EntityHandler(
     @DeleteMapping("/{entityId}")
     fun delete(@PathVariable entityId: String): Mono<ResponseEntity<*>> {
         return entityId.toMono()
-            .map {
+            .zipWith(extractSubjectOrEmpty())
+            .doOnNext {
+                if (!entityService.exists(entityId))

```

Comment:
now that you check the existence here, it probably changes the semantics of the returned response a few lines below

Reply:
What do you mean, I should throw an Exception as well ?

Reply:
Or maybe I could just have some mechanism to have the authorizationService answer something in this case. What do you think ?

Reply:
My point was this endpoint returns a 404 if deleteEntity returns 0, which is now unlikely to be a 404 because entity existence is checked here.

Reply:
In the ExceptionHandler ResourceNotFound exception is mapped to a 404 error, Or I still miss something..